src/analysis/syntax_doc.ml

```diff
@@ -1,12 +1,765 @@
+          Some
+            { name = "Zero-alloc opt annotation";
+              description = "Same as [@zero_alloc] in optimized builds only.";
+              documentation = doc_url;
+              level = Advanced
+            }
+        | "assume" ->
+          Some
+            { name = "Zero-alloc assume annotation";
+              description =
+                "This function is assumed to be \"zero_alloc\" but the \
+                 compiler does not guarantee it.";
+              documentation = doc_url;
+              level = Advanced
+            }
+        | "assume_unless_opt" ->
+          Some
+            { name = "Zero-alloc assume_unless_opt annotation";
+              description =
+                "Same as [@zero_alloc opt]  in optimized builds. Same as \
+                 [@zero_alloc assume] in non-optimized builds.";
+              documentation = doc_url;
+              level = Advanced
+            }
+        | "strict" ->
+          Some
+            { name = "Zero-alloc strict annotation";
+              description =
+                "This function does not allocate on the OCaml heap (both \
+                 normal and exception returns).";
+              documentation = doc_url;
+              level = Advanced
+            }
+        | "arity" ->
+          Some
+            { name = "Zero-alloc arity annotation";
+              description =
+                "The function does not allocate when applied to [n] arguments. \
+                 This can be used to override the arity inferred based on the \
+                 number of arrows in the type.";
+              documentation = doc_url;
+              level = Advanced
+            }
+        | _ -> None)
+      | _ ->
+        Some
+          { name = "Unrecognized zero-alloc annotation";
+            description = "This is an unrecognized zero-alloc annotation.";
+            documentation = doc_url;
+            level = Advanced
+          })
+    | { attr_name = { txt = "noalloc"; _ }; _ } ->
+      Some
+        { name = "Noalloc annotation";
+          description =
+            "This external does not allocate, does not raise exceptions, and \
+             does not release the domain lock. The compiler will optimize uses \
+             to a direct C call.";
+          documentation = syntax_doc_url Ocaml "intfc.html#ss:c-direct-call";
+          level = Advanced
+        }
+    (* Inlining annotations *)
+    | { attr_name = { txt = "inline"; _ }; attr_payload; _ } -> (
+      let inline_always_annot : syntax_info =
+        Some
+          { name = "Inline always annotation";
+            description =
+              "On a function declaration, causes the function to be inlined at \
+               all known call sites (can be overridden by [@inlined]).  In \
+               addition it will be made available for inlining in other source \
+               files (with appropriate build settings permitting .cmx file \
+               visibility)";
```

Comment:
```suggestion
              "This function will be inlined at \
               all call sites (can be overridden by [@inlined never])";
```

Several changes:
* Removed "known" from "all known call sites". Not sure what "known" is meant to communicate there.
* Changed `[@inlined]` to `[@inlined never]`, but please double-check this.
* Dropped the bit about build settings. Is that part necessary? If so, what exactly should the build settings be?